sympy/physics/control/lti.py

```diff
@@ -1422,3 +1441,520 @@ def _eval_rewrite_as_TransferFunction(self, num, den, **kwargs):
+    [   s + 2          p + s     ]
+
+    ``subs()`` also supports multiple substitutions.
+
+    >>> tfm_2.subs({p: 2, a: 1})  # substituting p with 2 and a with 1
+    TransferFunctionMatrix(((TransferFunction(s + 1, s**2 + s + 1, s), TransferFunction(-3, s + 2, s)), (TransferFunction(12, s + 2, s), TransferFunction(-s - 1, s**2 + s + 1, s)), (TransferFunction(3, s + 2, s), TransferFunction(-12, s + 2, s))))
+    >>> pprint(_, use_unicode=False)
+    [  s + 1        -3     ]
+    [----------    -----   ]
+    [ 2            s + 2   ]
+    [s  + s + 1            ]
+    [                      ]
+    [    12        -s - 1  ]
+    [  -----     ----------]
+    [  s + 2      2        ]
+    [            s  + s + 1]
+    [                      ]
+    [    3          -12    ]
+    [  -----       -----   ]
+    [  s + 2       s + 2   ]
+
+    Users can reduce the ``Series`` and ``Parallel`` elements of the matrix to ``TransferFunction`` by using
+    ``doit()``.
+
+    >>> tfm_6 = TransferFunctionMatrix([[Series(tf_3, tf_4), Parallel(tf_3, tf_4)]])
+    >>> tfm_6
+    TransferFunctionMatrix(((Series(TransferFunction(3, s + 2, s), TransferFunction(-a + p, 9*s - 9, s)), Parallel(TransferFunction(3, s + 2, s), TransferFunction(-a + p, 9*s - 9, s))),))
+    >>> pprint(tfm_6, use_unicode=False)
+    [ -a + p   3     -a + p     3  ]
+    [-------*-----  ------- + -----]
+    [9*s - 9 s + 2  9*s - 9   s + 2]
+    >>> tfm_6.doit()
+    TransferFunctionMatrix(((TransferFunction(-3*a + 3*p, (s + 2)*(9*s - 9), s), TransferFunction(27*s + (-a + p)*(s + 2) - 27, (s + 2)*(9*s - 9), s)),))
+    >>> pprint(_, use_unicode=False)
+    [    -3*a + 3*p     27*s + (-a + p)*(s + 2) - 27]
+    [-----------------  ----------------------------]
+    [(s + 2)*(9*s - 9)       (s + 2)*(9*s - 9)      ]
+    >>> tf_9 = TransferFunction(1, s, s)
+    >>> tf_10 = TransferFunction(1, s**2, s)
+    >>> tfm_7 = TransferFunctionMatrix([[Series(tf_9, tf_10), tf_9], [tf_10, Parallel(tf_9, tf_10)]])
+    >>> tfm_7
+    TransferFunctionMatrix(((Series(TransferFunction(1, s, s), TransferFunction(1, s**2, s)), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), Parallel(TransferFunction(1, s, s), TransferFunction(1, s**2, s)))))
+    >>> pprint(tfm_7, use_unicode=False)
+    [ 1      1   ]
+    [----    -   ]
+    [   2    s   ]
+    [s*s         ]
+    [            ]
+    [ 1    1    1]
+    [ --   -- + -]
+    [  2    2   s]
+    [ s    s     ]
+    >>> tfm_7.doit()
+    TransferFunctionMatrix(((TransferFunction(1, s**3, s), TransferFunction(1, s, s)), (TransferFunction(1, s**2, s), TransferFunction(s**2 + s, s**3, s))))
+    >>> pprint(_, use_unicode=False)
+    [1     1   ]
+    [--    -   ]
+    [ 3    s   ]
+    [s         ]
+    [          ]
+    [     2    ]
+    [1   s  + s]
+    [--  ------]
+    [ 2     3  ]
+    [s     s   ]
+
+    See Also
+    ========
+
+    TransferFunction, Series, Parallel, Feedback
+
+    """
+    def __new__(cls, arg):
```

Comment:
```python
>>> G = TransferFunction(s, s + 1, s)
>>> TransferFunctionMatrix([G])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/namannimmo/oss/sympy/sympy/physics/control/lti.py", line 1703, in __new__
    var = arg[0][0].var
TypeError: 'TransferFunction' object is not subscriptable
```
Raise an error mentioning `arg` can only be a nested list. Add test for that as well.